### ISSUE TITLE
Implement Loss Function from "Learning Deep Structure-Preserving Image-Text Embeddings"

### ISSUE THREAD

**Issue by alam-shahul** [jiayueg/multimodal#1]
Figure out how to define "neighbors" for batches of training datapoints. See https://arxiv.org/pdf/1511.06078.pdf
